src/components/BookForm.js

```diff
@@ -64,19 +70,25 @@ class BookForm extends React.Component {
     } else {
       value = e.currentTarget.value;
     }
-    this.props.editBook(name, value);
+    if (name === "isbn") {
+      this.props.checkForCollision(value);
+    }
+    if (value) {
+      book[name] = value;
+    } else {
+      delete book[name];
+    }
+    this.setState({ book });
   };
 
   render() {
-    const { ...book } = this.props.bookFormContent;
-    let classes = "modal";
-    if (this.props.showEditForm) classes += " is-visible";
+    const { ...book } = this.state.book;
```

Comment:
You can just do this here (although the spread operator is awesome): 

```suggestion
    const { book } = this.state;
```

Reply:
ooh smort!